### ISSUE TITLE
Trapdoor Config

### ISSUE THREAD

**Issue by heeylost** [TehBrian/BuildersUtilities#70]
Trapdoors can´t open if you have something in your hand. It would be nice open it in any case by config!

**Comment by RedstoneFuture** [comment]
That is correct. With wooden trapdoors, you can still open the trapdoors. Can we just do it like the wooden version - without config? The same for doors.

**Comment by TehBrian** [comment]
It's been a while since I looked into this, but I'm fairly sure the reason that there was an implementation difficulty in replicating wooden trapdoors' functionality. I'll take another look at this once I can get back to my computer, though

**Comment by TehBrian** [comment]
okay, finally got back to my computer lol

so the reason for the current behavior is summarized pretty nicely by this javadoc comment on the iron door listener

```java
/**
 * Allows players to open iron doors/trapdoors similarly to how wooden
 * doors/trapdoors can be opened.
 * <p>
 * The difference between this functionality and native (wooden) functionality is
 * that players <strong>must</strong> have an empty hand to toggle doors/trapdoors.
 * This is to prevent glitchy single-tick block placement (and subsequent
 * disappearance) because the client isn't aware of this functionality.
 */
```

essentially, because the client believes that iron doors cannot be toggled, clicking on one with a block in your hand will cause the block to be placed and then quickly disappear, which feels sorta janky

i initially didn't find that acceptable, but i think i've changed my mind. if builders want the jank-free experience, then they can still switch to an empty hand, but when building it's often inconvenient to do so, so the plugin should ideally get out of your way and let you do what you want, even if it's visually a bit glitchy
